rtps-r52/tests/mailbox.c

```diff
@@ -11,9 +12,14 @@
 
 // libhpsc
 #include <devices.h>
+#include <link-mbox.h>
+#include <link.h>
+#include <hpsc-msg.h>
 
 #include "test.h"
 
+#define SEND_RCV_TIME_OUT 1000000000000
```

Comment:
When building this code, I get the warning:

```
tests/mailbox.c:21:27: warning: unsigned conversion from 'long long int' to 'rtems_interval' {aka 'unsigned int'} changes value from '1000000000000' to '3567587328' [-Woverflow]
   21 | #define SEND_RCV_TIME_OUT 1000000000000
      |                           ^~~~~~~~~~~~~
tests/mailbox.c:66:22: note: in expansion of macro 'SEND_RCV_TIME_OUT'
   66 |                      SEND_RCV_TIME_OUT, write_buf, sizeof(write_buf),
```
The same warning applies to line 67 as well.  Please watch out for any warnings during the build.  In this case, `rtems_interval` is actually an unsigned int that can't store such a large integer.  Please verify that the number of ticks you're passing in is correct or, if it is, the `rtems_interval` type definition needs to change.

Reply:
Timeouts are specified in "ticks", so you need to know what those are before you can choose a reasonable value.  I don't recall off the top of my head how many clock cycles or microseconds a tick amounts to, but the RTEMS BSP has this information somewhere in it and there are some macros that can tell you the microseconds per tick (and vice versa) and convert between them.  In other tests we use timeouts of 1000 ticks or so, but you may need something longer if communicating with linux userspace.